docs/calendar-review.md

```diff
@@ -0,0 +1,212 @@
+
+```javascript
+function isLastDayOfYear(date) {
+  // Note: This is a BAD example!
+  return date.month === 12 && date.day === 31;
+}
+
+// works for ISO calendar
+isLastDayOfYear(Temporal.PlainDate.from('2019-12-31')); // => true
+isLastDayOfYear(Temporal.PlainDate.from('2020-01-01')); // => false
+
+// fails for some non-ISO calendars
+hebrewNewYearsEve = Temporal.PlainDate.from(
+  { year: 5780, monthCode: 'M12', day: 29, calendar: 'hebrew' });
+isLastDayOfYear(hebrewNewYearsEve); // => expected: true, actual: false
+```
+
+### Mitigating the Unexpected Calendar Problem
+
+There are two ways to avoid the Unexpected Calendar Problem: the preferred solution of writing calendar-safe code, or a fallback option to validate or coerce inputs to a known calendar.
+
+#### Writing Calendar-Safe Code
+
+The best way to mitigate the Unexpected Calendar Problem is to write “calendar-safe” code that works for all built-in calendars. For example, the function above can easily be written to be calendar-safe by using the `monthsInYear` and `daysInMonth` properties instead of hard-coding ISO constants. Developers who follow documented [best practices for writing calendar-safe Temporal code](https://tc39.es/proposal-temporal/docs/calendar.html#writing-cross-calendar-code) will be able to use the same code for all built-in calendars.
+
+Beyond documentation, the Temporal API itself has been designed to make it easier to write calendar-safe code. For example:
+
+*   All built-in calendars (including ISO) use the same set of date fields: `year`, `month`, `day`, and `monthCode`. These fields act the same way in all built-in calendars.
+*   The `month` property is a continuous (no gaps), 1-based index of the month in the current year. This aligns with ISO behavior and avoids unexpected behavior from lunisolar calendars where each year may have a different set of months.  A separate `monthCode` string field is used for a year-independent month identifier.
+*   The `year` property is a signed value that’s relative to a specific calendar era. This matches ISO calendar behavior and prevents bugs from, for example, eras like B.C. where years count backwards.  Separate `eraYear` and `era` fields are used for measuring years within the context of a specific calendar era.
+*   Differences between calendars are exposed via convenience properties available in every calendar, e.g. `monthsInYear`, `daysInMonth`, `inLeapYear`, etc. Userland code can completely avoid the use of calendar-specific constants (e.g. 12 months per year) by using these properties. 
+*   There is never an implicit default calendar system. Developers must explicitly decide whether they’re using the ISO calendar or another calendar. For example, shortcut methods on the `Temporal.now` object are named with an ISO prefix to avoid ambiguity, such as `Temporal.now.zonedDateTimeISO()`.
```

Comment:
```suggestion
*   There is never an implicit default calendar system. Developers must explicitly decide whether they're using the ISO calendar or another calendar. For example, shortcut methods on the `Temporal.now` object are named with an ISO prefix to avoid ambiguity, such as `Temporal.now.zonedDateTimeISO()`.
```